cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -87,6 +97,49 @@ public MetadataSearchTargetType apply(String input) {
         return MetadataSearchTargetType.valueOf(input.toUpperCase());
       }
     };
+  private static final Comparator<MetadataSearchResultRecord> NAME_ASCENDING_COMPARATOR =
+    new Comparator<MetadataSearchResultRecord>() {
+      @Override
+      public int compare(MetadataSearchResultRecord o1, MetadataSearchResultRecord o2) {
+        NamespacedEntityId entityId1 = o1.getEntityId();
+        NamespacedEntityId entityId2 = o2.getEntityId();
+        int compareResult = entityId1.getEntityName().compareTo(entityId2.getEntityName());
+        if (compareResult != 0) {
+          return compareResult;
+        }
+        // same name, but different entity types?
+        compareResult = entityId1.getEntityType().compareTo(entityId2.getEntityType());
+        if (compareResult != 0) {
+          return compareResult;
+        }
+        // same name and entity type, but different namespace?
+        compareResult = entityId1.getNamespace().compareTo(entityId2.getNamespace());
+        if (compareResult != 0) {
+          return compareResult;
+        }
+        // same name, entity type and namespace, so dig deeper
+        switch (entityId1.getEntityType()) {
```

Comment:
yup, if they were different, line 113 would have been executed, and control would not reach here.